app/models/person.rb

```diff
@@ -232,8 +214,16 @@ def self.username_blacklist
     USERNAME_BLACKLIST
   end
 
-  def self.username_available?(username)
-     !Person.find_by_username(username).present? && !username.in?(USERNAME_BLACKLIST)
+  def self.username_available?(username, community_id)
+    if FeatureFlag.where(community_id: community_id, feature: :new_login, enabled: true).present?
```

Comment:
This is accessing the model directly. What if you add this to the Service layer? Something like this:

```ruby
    def enabled?(community_id:, feature:)
      Result::Success.new(@feature_flag_store.get(community_id).include?(feature))
    end
```